src/formDesigner/components/ConceptDetails.js

```diff
@@ -19,12 +19,75 @@ import UserInfo from "../../common/model/UserInfo";
 import { connect } from "react-redux";
 import { Privilege } from "openchs-models";
 
-function ConceptDetails({ userInfo, ...props }) {
+function ConceptDetails({ subjectType, formMappings, userInfo, ...props }) {
   const [editAlert, setEditAlert] = useState(false);
   const [data, setData] = useState({});
   const [usage, setUsage] = useState({});
   const [addressLevelTypes, setAddressLevelTypes] = useState([]);
   const [subjectTypeOptions, setSubjectTypeOptions] = React.useState([]);
+  const [fdata, setFdata] = useState({});
+  const [idToNameConverter, setIdToNameConverter] = useState({});
+
+  const fetchFormData = async formUuid => {
+    try {
+      const response = await http.get(`/forms/export?formUUID=${formUuid}`);
+      const form = response.data;
+      const formDetails = {};
+      const idToName = {};
+      form.formElementGroups.forEach(feg => {
+        feg.formElements.forEach(fe => {
+          const concept = fe.concept;
+          if (!feg.voided && !fe.voided) {
+            if (!formDetails[form.name]) {
+              formDetails[form.name] = [];
+            }
+            const elementDetails = {
+              formName: form.name,
+              sectionName: feg.name,
+              questionName: fe.name,
+              parentUUID: fe.parentFormElementUuid,
+              conceptName: concept.name,
+              dataType: concept.dataType
+            };
+            formDetails[form.name].push(elementDetails);
+            idToName[fe.uuid] = fe.name;
+          }
+        });
+      });
+
+      setFdata(prevFdata => ({
+        ...prevFdata,
+        ...formDetails
+      }));
+      setIdToNameConverter(prevdata => ({
+        ...prevdata,
+        ...idToName
+      }));
+    } catch (error) {
+      console.error("Error fetching form data:", error);
+      alert("Failed to fetch form data. Please try again later.");
+    }
+  };
+
+  useEffect(() => {
+    const fetchUsageAndFormDetails = async () => {
+      try {
+        const usageResponse = await http.get(`/web/concept/usage/${props.match.params.uuid}`);
+        setUsage(usageResponse.data);
+
+        if (usageResponse.data.forms) {
+          usageResponse.data.forms.forEach(form => {
+            fetchFormData(form.formUUID);
```

Comment:
instead of doing this, why not enhance the /web/concept/usage url to give you all the data instead?